eventlet/dagpool.py

```diff
@@ -0,0 +1,484 @@
+        own key, which will cause its return value to be discarded.
+
+        Calling post(key, value, replace=False) (the default 'replace') when a
+        value for that key has already been posted, by any means, raises
+        Collision.
+
+        Calling post(key, value, replace=True) when a value for that key has
+        already been posted, by any means, replaces the previously-stored
+        value. However, that may make it complicated to reason about the
+        behavior of greenthreads waiting on that key.
+
+        After a post(key, value1) followed by post(key, value2, replace=True),
+        it is unspecified which pending wait_each([key...]) calls (or
+        greenthreads iterating over 'results' involving that key) will observe
+        value1 versus value2. It is guaranteed that subsequent
+        wait_each([key...]) calls (or greenthreads spawned after that point)
+        will observe value2.
+        """
+        # First, check if we're trying to post() to a key with a running
+        # greenthread.
+        try:
+            coro = self.coros[key]
+        except KeyError:
+            # If there's no running greenthread with this key, carry on.
+            pass
+        else:
+            # A DAGPool greenthread is explicitly permitted to post() to its
+            # OWN key.
+            if coro.greenthread is not greenthread.getcurrent():
+                # oh oh, trying to post a value for running greenthread from
+                # some other greenthread
+                raise Collision(key)
+
+        # Here, either we're posting a value for a key with no greenthread or
+        # we're posting from that greenthread itself.
+
+        # Has somebody already post()ed a value for this key?
+        try:
+            self.values[key]
+        except KeyError:
+            # no, we're good
+            pass
+        else:
+            # unless replace == True, this is a problem
+            if not replace:
+                raise Collision(key)
+
+        # Either we've never before posted a value for this key, or we're
+        # posting with replace == True.
+
+        # update our database
+        self.values[key] = value
+        # and wake up pending waiters
+        self.event.send()
+        # The comment in Event.reset() says: "it's better to create a new
+        # event rather than reset an old one". Okay, fine. We do want to be
+        # able to support new waiters, so create a new Event.
+        self.event = Event()
+
+    def __getitem__(self, key):
+        """
+        __getitem__(key) (aka dagpool[key]) blocks until 'key' has a value,
+        then delivers that value.
+        """
+        # This is a degenerate case of wait_each(). Construct a list
+        # containing only this 'key'. list(wait_each()) will make a list
+        # containing exactly one (key, value) pair. Extract that pair, then
+        # extract just its value.
+        return list(self.wait_each([key]))[0][1]
```

Comment:
If you don't modify the list, then `tuple` seems a better fit.
And then you only use first item, so no need to construct any sequence storage, just fetch first pair from iterator using `next()` or `for-break`.

```
for _, value in self.wait_each( (key,) ):
  return value
```